File: Stacks/NumberOfVisiblePeople/visiblePeople.py
from typing import List

def canSeePersonsCount(self, heights: List[int]) -> List[int]:
    n = len(heights)
    ans = [0] * n
    st = [heights[n - 1]]

    for i in range(n - 2, -1, -1):
        count = 0

        while len(st) and heights[i] >= st[-1]:
            st.pop()
            count += 1

        if len(st):
            count += 1

        ans[i] = count
        st.append(heights[i])

    return ans

File: Stacks/NumberOfVisiblePeople/visiblePeople.cpp
#include <vector>
#include <stack>
using namespace std;

class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& nums) {
        int n = nums.size();
        vector<int>ans(n);
        ans[n-1]=0;
        stack<int>st;
        st.push(nums[n-1]);
        for(int i=n-2;i>=0;i--){
            int count = 0;
            while(st.size() && nums[i]>st.top()){
                st.pop();
                count++;
            }
            if(st.size() && nums[i]<st.top())count++;
            ans[i]=count;
            st.push(nums[i]);
        }
        return ans;
    }
};